### DISCUSSION TITLE
Tools to manage streams

### DISCUSSION THREAD

**Discussion by wrobell** [rabbitmq/rabbitmq-server#3998] [Other] [answered]
`rabbitmqctl` allows to list and delete queues and thus the streams. But it seems it does not allow to create or manage the streams, i.e. to set retention policy?
 
The Java client has a [performance tool](https://rabbitmq.github.io/rabbitmq-stream-java-client/stable/htmlsingle/#the-performance-tool), which allows to create and manage streams.

Is there a plan to extend `rabbitmqctl` with appropriate streams functionality, or shall each library provide its own tooling?

**Comment by acogoluegnes** [comment, marked as answer]
Streams are queues in RabbitMQ resource management mechanism, so as long as you can create queues with their arguments, you can create a stream (just like a quorum queue). Same thing with listing and deleting them (you should see the streams with `rabbitmqctl list_queues`).

**Reply by wrobell** [reply]
Thanks for the tip.

If anyone is looking for a command, an example
```
$ rabbitmqadmin declare queue name=some-queue queue_type=stream arguments='{"x-max-length-bytes": 1000000, "x-stream-max-segment-size-bytes": 100000, "x-max-age": "5m"}'
```

[RabbitMQ management pluging](https://www.rabbitmq.com/management.html) is needed.
